### ISSUE TITLE
Dark theme styles and colours

### ISSUE THREAD

**Issue by glocalglocal** [writingtool-org/writingtool#26]
It would be useful to be able to change underline styles and colours beyond the three presets. In dark themes with light fonts on dark background they are hardly visible. More presets may be sufficient. Overall, LO is not great with dark themes and Writingtool adds to the problem at the moment.

**Comment by glocalglocal** [comment]
I didn't realise I had to wait until tomorrow and installed the current nightly. I will try again tomorrow but in the meanwhile the current nightly is not working at all. Menu options under Tools do nothing, there are no errors shown in text, and WritingTool doesn't appear under Writing Aids any more. Tomorrow's nightly may be like that as well.

**Comment by FredKruse** [comment]
Are you using Java 17 or higher? WritingTool 1.1 snapshots require it (newer LanguageTool versions require it).

**Comment by glocalglocal** [comment]
I reinstalled java and WritingTool is back. Thanks! Awaiting the nightly tomorrow.

**Comment by glocalglocal** [comment]
These preselected dark theme colours work a little better but I think they should be lighter, esp green. 

Users can set their dark background to different shades of grey so, ideally, they would be able to pick their own Writingtool colours too, but that may not be a priority now.

**Comment by glocalglocal** [comment]
Below is what I see (yes, there is a blue wave somewhere there). I think blue and green could be made to stand out more, like red really. If dark mode is meant to reduce eye strain for people who don't like brightness (like myself), trying to spot the wavy lines increases it. But it's OK for now and I can wait until 1.2. Many thanks.  

My dark theme setup:
![image](https://github.com/user-attachments/assets/b4e64e3b-cfa4-40fb-8a9e-68f72763b7d7)

LO's default dark mode: 
![image](https://github.com/user-attachments/assets/8b012df7-beb6-47db-a910-9f2883ac4d53)

**Comment by glocalglocal** [comment]
That's great! Many thanks. It looks much better now: 
![image](https://github.com/user-attachments/assets/b8f56391-cd71-4073-926d-48df49b7ffcc)

Two issues: it took several attempts and Refresh for my selected colours to stick. Later again WT put its defaults back as Custom colours even during the same session. It also loses Recent colours.  

Also, sometimes OK on the palette dialogue is not clickable. I click Cancel but then I find that the colour I was trying to select was in fact registered correctly. Then, I return to my document and WT has returned to its defaults as above.

**Comment by FredKruse** [comment]
The issue should be solved. Please test it tomorrow and give me feedback.

**Comment by glocalglocal** [comment]
Installed the one dated 21/12. Options and About do nothing now so I can't check the colours.

**Comment by glocalglocal** [comment]
Today, after a system restart I can get into Options but colours are forgotten again. I think that once the colours are set successfully, they stick unless I interfere with one of the colours.  

Oh, WT's dialogues should also follow the dark theme applied to LO's menus and dialogues. Currently, Check brings up a painfully bright dialogue.

**Comment by FredKruse** [comment]
Sometimes old processes stay alive after shutdown of LO.  This can lead to WritingTool not behaving properly. Please test the current version after a restart. Are the colors really not saved? I tried this several times, and it worked (after I fixed a bug in the 12/20 version).

The other point, changing the background colors of the dialogs with the LO theme, is a suggestion for further development. I have opened issue #29 for this. However, this will take a while and I don't know if I can install an automatic mechanism (depends on whether there is a corresponding interface to LO).

**Comment by glocalglocal** [comment]
Yes, the 23/12 version remembers colours. Recent colours is populated while the dialogue is open but it is blanked out when I return to the dialogue -- ie recent colours are forgotten.

**Comment by FredKruse** [comment]
The option dialog must be closed with OK. Otherwise, the colors will not be saved. The philosophy is that a Cancel restores the old state.
Please test it with the last version and close the issue if it works.

**Comment by glocalglocal** [comment]
Whatever I do, WT never remembers Recent colours when I reopen the dialogue. It hasn't done so once. Not a huge problem and probably a separate issue. Otherwise, colours work OK. Shall I close the issue?

**Comment by FredKruse** [comment]
The bug in the dark theme library has been fixed. The new snapshot will be available tomorrow. Please test it and give me feedback.

**Comment by glocalglocal** [comment]
Not sure what I should be looking for, but colours work OK.
